bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/ReadmeHelper.java

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.binding.miio.internal;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.openhab.binding.miio.internal.basic.MiIoBasicChannel;
+import org.openhab.binding.miio.internal.basic.MiIoBasicDevice;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * Support creation of the readme
+ *
+ * @author Marcel Verpaalen - Initial contribution
+ */
+public class ReadmeHelper {
+    private final static Logger logger = LoggerFactory.getLogger(ReadmeHelper.class);
+
+    public static void main(String[] args) {
+        ReadmeHelper rm = new ReadmeHelper();
+        System.out.println("## Creating device list\n");
+        StringWriter deviceList = rm.deviceList();
+        rm.checkDatabaseEntrys();
+        // System.out.println(deviceList);
```

Comment:
If you do have the time it would be nice if you could quickly check the commented out lines of code and remove them if possible.

If you do that and make the SAT happy we can merge this.